pkg/reconciler/pipelinerun/resources/pipelinerunstate.go

```diff
@@ -132,16 +132,16 @@ func (state PipelineRunState) IsBeforeFirstTaskRun() bool {
 func (state PipelineRunState) AdjustStartTime(unadjustedStartTime *metav1.Time) *metav1.Time {
 	adjustedStartTime := unadjustedStartTime
 	for _, rpt := range state {
-		if rpt.TaskRun == nil {
-			if rpt.RunObject != nil {
-				creationTime := rpt.RunObject.GetObjectMeta().GetCreationTimestamp()
+		if len(rpt.TaskRuns) == 0 {
```

Comment:
I don't think a nested conditional is needed here. This could be simplified to
```
if len(rpt.RunObjects) > 0 { ... } 
else if len(rpt.TaskRuns) > 0 { ... } 
```